coalib/results/TextPosition.py

```diff
@@ -15,12 +19,21 @@ def __init__(self, line: (int, None)=None, column: (int, None)=None):
         :param line:        The line in file or None, the first line is 1.
         :param column:      The column indicating the character. The first one
                             in a line is 1.
+        :param text:        The text containing the character.
+        :param position:    Position identifying the index of character in text.
         :raises TypeError:  Raised when line or columns are no integers.
         :raises ValueError: Raised when a column is set but line is None.
         """
         if line is None and column is not None:
             raise ValueError("A column can only be set if a line is set.")
 
+        if line and column and text and not position:
+            position = TextPosition.calc_position(text, line, column)
+
+        if (line and column) is None and (text and position) is not None:
+            line, column = TextPosition.calc_line_col(text, position)
```

Comment:
I am not sure how to explain this in a better way so I'm just going to do it with code.

```
@property
def position(self):
    if self.text is None:
        return None
    if self._position is None:
        # _position is calculated lazily. i.e. ONLY when position is asked (for the first time)
        # If a user never tries to use `position` it is never computed.
        self._position = self.calc_position(self.text, self.line, self.column)
    return self._position
```